### PR TITLE
modify ModelManifest.

### PR DESCRIPTION
**Pull request by vwxyzh**
@chenkennt @ansyral @hellosnow @superyyrrzz @qinezh 
#449

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.Engine/HostService.cs [hunk 1]
```diff
@@ -548,18 +558,23 @@ public void SaveIntermediateModel(string intermediateFolder, ModelManifest lmm,
...
+        public FileModel LoadIntermediateModel(string intermediateFolder, ModelManifest lmm, string fileName)
         {
             var processor = Processor as ISupportIncrementalDocumentProcessor;
             if (processor == null)
             {
                 return null;
             }
-            using (var stream = new FileStream(fileName, FileMode.Open))
+            string lfn;
+            if (!lmm.Models.TryGetValue(fileName, out lfn))
```
Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.Engine/HostService.cs]: >lmm [](start = 17, length = 3)

check lmm != null

File: src/Microsoft.DocAsCode.Build.Engine/HostService.cs [hunk 2]
```diff
@@ -523,21 +524,30 @@ where condition(f)
...
                     if (lmm == null)
                     {
                         throw new InvalidDataException($"Full build hasn't loaded model {pair.Key.FullPath}");
                     }
-                    File.Copy(Path.Combine(intermediateFolder, lmm.BaseDir, fileName), Path.Combine(intermediateFolder, cmm.BaseDir, fileName));
+                    string lfn;
+                    if (!lmm.Models.TryGetValue(pair.Key.File, out lfn))
+                    {
+                        throw new InvalidDataException($"Last build hasn't loaded model {pair.Key.FullPath}");
+                    }
```
Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.Engine/HostService.cs]: can union with the check "lmm == null"?

**Comment by ansyral** [comment]
> 

remove?

---
Refers to: src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs:683 in c86118c. [](commit_id = c86118c966bc232448a6c9f65fd8203670dd8242, deletion_comment = False)

**Comment by ansyral** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Build.Engine/HostService.cs (hunk 2 above)
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.Engine/HostService.cs]: Message is different.

---
In reply to: [78681597](https://github.com/dotnet/docfx/pull/698#discussion_r78681597) [](ancestors = 78681597)

**Pull request merged**